### PR TITLE
k8s: add deletion to Setup

### PR DESCRIPTION
**Pull request by kopiczko**

### CODE REVIEW

File: k8s/setup.go [hunk 1]
```diff
@@ -80,3 +80,35 @@ func (s *Setup) EnsureNamespace(ctx context.Context, namespace string) error {
...
+	s.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensuring deletion of Kubernetes Namespace %#q", namespace))
+
+	o := func() error {
+		{
+			n := &v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: namespace,
+				},
+			}
+			_, err := s.k8sClient.CoreV1().Namespaces().Delete(n)
```
Comment (MarcelMue) [inline k8s/setup.go]: This seems to be incorrect. You are missing the `opts` in `Delete`.

**Comment by xh3b4sd** [comment]
Since this is a breaking change we should have documentation and knowledge sharing about it as well as a plan to roll it out. When you take care anyway then all good. Just want to raise awareness that this breaks code using the lib already.

**Comment by kopiczko** [comment]
> Since this is a breaking change we should have documentation and knowledge sharing about it as well as a plan to roll it out.

This was introduced like yesterday?
I will just do vendoring.

**Comment by xh3b4sd** [comment]
Yes it is all good. I just wanted to emphazise some consciousness of what we do here because we broke a lot in the past due to similar things.

**Comment by kopiczko** [comment]
Fishing for LGTM.

**Review by indyfree — APPROVED** [review]
LGTM, But not deep into the namespace setup thingies.

**Comment by kopiczko** [comment]
This is used to create/delete `giantswarm` namespace in e2e control plane. Possibly other namespaces in kvm-operator e2e setup.

**Comment by kopiczko** [comment]
I'm going with that. Feel free to comment if something needs improvements. I want to do vendoring right away.

**Pull request merged**